### DISCUSSION TITLE
Planned support for radio buttons

### DISCUSSION THREAD

**Discussion by scottstarsman** [pymupdf/PyMuPDF#1534] [General]
Is there a plan to support radio buttons in the near future? Thanks

**Comment by JorjMcKie** [comment]
It is not on my list currently, I'm afraid.
It would be another feature, which I need to create myself, because MuPDF has no support for it yet (last time I looked).
There are already quite a few things in this area which I added on top - and, regularly after each new MuPDF version, they cause significant effort to get them going again.

**Reply by scottstarsman** [reply]
Thanks for your fast reply. Is there some raw data available from the parsed document that I could use to extract this info?

Other than that, this has proven to be the best PDF tools I've used. Thanks for your work!

**Reply by JorjMcKie** [reply]
Thank you!

> raw data available from the parsed document 

You can walk through a page's annots, fields, links and then inspect their respective object definition, i.e. its xref "source". There always will exist things like `annot.xref`, `field.xref` etc. E.g.
```python
>>> for field in page.widgets():
	if field.field_type_string == "RadioButton":
		print(doc.xref_object(field.xref))
```
The responses look like this:
```postscript
<<
  /A 2228 0 R
  /AA <<
    /Bl 2230 0 R
  >>
  /AP <<
    /D <<
      /Off 2122 0 R
      /The#20provider#20or#20supplier#20is#20not#20enrolled#20with#20Medicare 2123 0 R
    >>
    /N <<
      /Off 2120 0 R
      /The#20provider#20or#20supplier#20is#20not#20enrolled#20with#20Medicare 2121 0 R
    >>
  >>
  /AS /Off
  /BS <<
    /S /S
    /W 1
  >>
  /DA (/ZaDb 10 Tf 0 g)
  /F 4
  /FT /Btn
  /Ff 49152
  /MK 2229 0 R
  /P 2111 0 R
  /Rect [ 36 466.06 45.36 476.14 ]
  /StructParent 717
  /Subtype /Widget
  /T (RadioButton3)
  /TU (The provider or supplier is not enrolled with Medicare)
  /Type /Annot
>>
```

**Reply by scottstarsman** [reply]
Excellent! I'll give that a try and see if I can get those fields. Thanks again for your help.
